sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/AvroCarbonWriter.java

```diff
@@ -823,6 +834,30 @@ public void write(Object object) throws IOException {
     }
   }
 
+  /**
+   * Load data of all avro files at given location iteratively.
+   *
+   * @throws IOException
+   */
+  @Override
+  public void write() throws IOException {
+    if (this.dataFiles == null || this.dataFiles.length == 0) {
+      throw new RuntimeException("'withAvroPath()' must be called to support loading avro files");
+    }
+    Arrays.sort(this.dataFiles);
+    for (File dataFile : this.dataFiles) {
+      this.loadSingleFile(dataFile);
+    }
+  }
+
+  private void loadSingleFile(File file) throws IOException {
+    DataFileReader<GenericData.Record> avroReader = SDKUtil.buildAvroReader(String.valueOf(file));
```

Comment:
Change `String.valueOf(file)` to `file.getPath()`

Reply:
done